lib/IDE/TypeReconstruction.cpp

```diff
@@ -748,6 +748,27 @@ static void VisitNodeAssociatedTypeRef(
           return;
         }
       }
+      ProtocolType *protocol_type = type->getAs<ProtocolType>();
+      if (protocol_type) {
+        Identifier assoc_id = ast->getIdentifier(ident->getText());
+        ProtocolDecl *protocol_decl = protocol_type->getDecl();
+        AssociatedTypeDecl *associated = nullptr;
+        for (auto member : protocol_decl->getMembers()) {
+          auto *atd = dyn_cast<AssociatedTypeDecl>(member);
+          if (!atd)
+            continue;
+          if (assoc_id == atd->getName()) {
+            associated = atd;
+            break;
+          }
+        }
+        if (associated) {
+          result._types.push_back(
+          DependentMemberType::get(protocol_type, assoc_id));
```

Comment:
Indentation is wrong here. Also you can’t have a DependentMemberType with an existential base. When does such a mangling come up?

Reply:
It shows up as type in DWARF (on some project I don't have access to anymore). You think we should just reject this outright?

Reply:
Unless you can come up with a test case, in which case we need to see why it’s mangling such a type.

Reply:
Makes sense, I'll send a new pull request soon. Thanks for your feedback!